### PR TITLE
fix: don't repeat the same stee sources field

### PR DESCRIPTION
**Pull request by y-okumura-isp**
Signed-off-by: y-okumura-isp <y-okumura@isp.co.jp>

### CODE REVIEW

File: src/tilde/test/test_stee_node.cpp [hunk 1]
```diff
@@ -536,3 +536,89 @@ TEST_F(TestSteeNode, two_topics_one_pub)
   rclcpp::spin_some(checker_node);
   EXPECT_TRUE(checker_received);
 }
+
+TEST_F(TestSteeNode, remove_duplicated_sources)
```
[the remark this answers is not in the document]
Comment (y-okumura-isp) [reply src/tilde/test/test_stee_node.cpp:540]: looks clang problem

File: src/tilde/test/test_stee_source_table.cpp [hunk 2]
```diff
@@ -20,9 +20,54 @@
 #include <utility>
 #include <vector>
 
+using tilde::SteeSourceCmp;
 using tilde::SteeSourcesTable;
 using tilde_msg::msg::SteeSource;
 
+TEST(TestSteeSourceCmp, simple_test)
```
[the remark this answers is not in the document]
Comment (y-okumura-isp) [reply src/tilde/test/test_stee_source_table.cpp:27]: looks google test problem

File: src/tilde/test/test_stee_source_table.cpp [hunk 3]
```diff
@@ -279,3 +324,46 @@ TEST_F(TestSteeSourcesTable, stamp_deletion)
   // check topic1 msg1 is purged
   EXPECT_EQ(table.get_sources("topic1", topic11_stamp).size(), 0u);
 }
+
+TEST_F(TestSteeSourcesTable, duplicated_stamp)
```
[the remark this answers is not in the document]
Comment (y-okumura-isp) [reply src/tilde/test/test_stee_source_table.cpp:328]: google-test problem

File: src/tilde/test/test_stee_node.cpp [hunk 4]
```diff
@@ -536,3 +536,89 @@
...
+  auto node_c = std::make_shared<SteeNode>("nodeC", options);
+  auto checker_node = std::make_shared<rclcpp::Node>("checker_node", options);
+
+  // nodeA directly publish SteePointCloud2 to fill sources field manually
+  auto pub_a = node_a->create_publisher<SteePointCloud2>("a/stee", 1);
+
+  // subscriptions of A
+  bool get_b_a{false};
+  auto sub_b_a = node_b->create_stee_subscription<PointCloud2>(
+    "a", 1, [&get_b_a](std::shared_ptr<const PointCloud2> msg) {
```
[the remark this answers is not in the document]
Comment (y-okumura-isp) [reply src/tilde/test/test_stee_node.cpp:565]: #63

File: src/tilde/test/test_stee_node.cpp [hunk 5]
```diff
@@ -536,3 +536,89 @@
...
+      (void)msg;
+      get_b_a = true;
+    });
+  bool get_c_a{false};
+  auto sub_c_a = node_c->create_stee_subscription<PointCloud2>(
+    "a", 1, [&get_c_a](std::shared_ptr<const PointCloud2> msg) {
```
[the remark this answers is not in the document]
Comment (y-okumura-isp) [reply src/tilde/test/test_stee_node.cpp:571]: #63

File: src/tilde/test/test_stee_node.cpp [hunk 6]
```diff
@@ -536,3 +536,89 @@
...
+      (void)msg;
+      get_c_a = true;
+    });
+
+  auto pub_b = node_b->create_stee_publisher<PointCloud2>("b", 1);
+
+  auto pub_c = node_c->create_stee_publisher<PointCloud2>("c", 1);
+  bool get_c_b{false};
+  auto sub_c_b = node_c->create_stee_subscription<PointCloud2>(
+    "b", 1, [&pub_c, &get_c_b](std::shared_ptr<const PointCloud2> msg) {
```
[the remark this answers is not in the document]
Comment (y-okumura-isp) [reply src/tilde/test/test_stee_node.cpp:581]: #63

File: src/tilde/test/test_stee_node.cpp [hunk 7]
```diff
@@ -536,3 +536,89 @@
...
+      get_c_b = true;
+      pub_c->publish(*msg);
+    });
+
+  bool checker_node_called{false};
+  auto sub_checker = checker_node->create_subscription<SteePointCloud2>(
+    "c/stee", 1, [&checker_node_called](std::shared_ptr<const SteePointCloud2> msg) {
```
[the remark this answers is not in the document]
Comment (y-okumura-isp) [reply src/tilde/test/test_stee_node.cpp:588]: #63

File: src/tilde/include/tilde/stee_publisher.hpp [hunk 8]
```diff
@@ -181,10 +181,16 @@ class SteePublisher
 
   void set_sources(ConvertedMessageT * converted_msg)
   {
+    std::set<tilde_msg::msg::SteeSource, SteeSourceCmp> found;
+
     if (!is_explicit) {
```
[the remark this answers is not in the document]
Comment (y-okumura-isp) [reply src/tilde/include/tilde/stee_publisher.hpp]: fixed

File: src/tilde/src/stee_sources_table.cpp [hunk 9]
```diff
@@ -17,6 +17,7 @@
 #include <cassert>
 #include <map>
```
[the remark this answers is not in the document]
Comment (y-okumura-isp) [reply src/tilde/src/stee_sources_table.cpp:20]: fixed

[the remark this answers is not in the document]
Comment (y-okumura-isp) [reply src/tilde/src/stee_sources_table.cpp:20]: fixed

File: src/tilde/test/test_stee_source_table.cpp [hunk 10]
```diff
@@ -20,9 +20,54 @@
...
 #include <vector>
 
+using tilde::SteeSourceCmp;
 using tilde::SteeSourcesTable;
 using tilde_msg::msg::SteeSource;
 
+TEST(TestSteeSourceCmp, simple_test)
+{
+  SteeSourceCmp cmp;
+  tilde_msg::msg::SteeSource lhs, rhs;
```
[the remark this answers is not in the document]
Comment (y-okumura-isp) [reply src/tilde/test/test_stee_source_table.cpp]: fixed

**Review by nabetetsu — APPROVED** [review]
LGTM :+1:

**Pull request merged**